R/find_exports.R

```diff
@@ -2,18 +2,10 @@ find_exports <- function(clean_lns) {
   ids <- find_extendr_attrs_ids(clean_lns)
   start <- ids
   end <- dplyr::lead(ids, default = length(clean_lns) + 1L) - 1L
-  purrr::map2_dfr(
-    start,
-    end,
-    ~ extract_meta(clean_lns[seq(.x, .y, by = 1L)])
-  ) %>%
-    # Keeps only name, type (fn|impl) and lifetime of impl
-    # if present.
-    dplyr::transmute(
-      .data$name,
-      type = dplyr::if_else(is.na(.data$impl), "fn", "impl"),
-      .data$lifetime
-    )
+
+  purrr::map2_dfr(start, end, ~ extract_meta(clean_lns[.x:.y])) %>%
```

Comment:
This is not a major problem, but it may be worth replacing the `map_df` functions with other functions since they are marked as "Superseded" in purrr 1.0.
https://purrr.tidyverse.org/reference/map_dfr.html

Reply:
You know what isn't superseded? for loops! ;) 
I'll change to `purrr::map()` (though this is a great use case for `base::Map()` too! 🫣)

If we want to go with the recommended approach we would use `purrr::list_rbind()` however this would require us to have a hard requirement on `purrr (>= 1.0.0)`  see [release notes](https://github.com/tidyverse/purrr/releases/tag/v1.0.0). Otherwise we'll have to `do.call(rbind, ...)`. How do y'all want to proceed here? 

@Ilia-Kosenkov @multimeric

Reply:
Not part of this PR, leave it as is. `map_dfr` will not be removed from the package, but only receive critical bugfixes, there is no problem in using it until we resolve it. You can file a separate issue to track this.